src/include/network/lan/address.rb

```diff
@@ -632,93 +629,72 @@ def InitSlave(_key)
       # reorder the items
       l2 = []
       l1 = []
-      Builtins.foreach(
-        Convert.convert(items, from: "list", to: "list <term>")
-      ) do |t|
-        if Builtins.contains(
-          Ops.get_list(@settings, "SLAVES", []),
-          Ops.get_string(t, [0, 0], "")
-          )
-          l1 = Builtins.add(l1, t)
+      items.each do |t|
+        if @settings["SLAVES"].include? Ops.get_string(t, [0, 0], "")
+          l1 << t
         else
-          l2 = Builtins.add(l2, t)
+          l2 << t
         end
       end
 
       items = []
-      Builtins.foreach(Ops.get_list(@settings, "SLAVES", [])) do |s|
-        Builtins.foreach(
-          Convert.convert(l1, from: "list", to: "list <term>")
-        ) do |t|
-          items = Builtins.add(items, t) if Ops.get_string(t, [0, 0], "") == s
+      @settings["SLAVES"].each do |s|
+        l1.each do |t|
+          items << t if Ops.get_string(t, [0, 0], "") == s
         end
       end
 
-      items = Builtins.union(items, l2)
+      items += l2.sort_by { |t| justify_dev_name(t[0][0]) }
+
       UI.ChangeWidget(:msbox_items, :Items, items)
       enableSlaveButtons
 
       nil
     end
 
+    # A helper for sort devices by name. It justify at right with 0's numeric parts of given
+    # device name until 5 digits.
+    #
+    # ==== Examples
+    #
+    #   justify_dev_name("eth0") # => "eth00000"
+    #   justify_dev_name("eth111") # => "eth00111"
+    #   justify_dev_name("enp0s25") # => "enp00000s00025"
+    #
+    # @param name [String] device name
+    # @return [String] given name with numbers justified at right
+    def justify_dev_name(name)
+      splited_dev_name = name.scan(/\p{Alpha}+|\p{Digit}+/)
```

Comment:
np: Maybe in this case it does not matter, but I like to use `\A` and `\Z` to match the beginning at the end.